static/app/views/settings/organizationDeveloperSettings/sentryFunctionDetails.tsx

```diff
@@ -0,0 +1,57 @@
+import {RouteComponentProps} from 'react-router';
+
+import {addLoadingMessage} from 'sentry/actionCreators/indicator';
+import Form from 'sentry/components/forms/form';
+import JsonForm from 'sentry/components/forms/jsonForm';
+import FormModel from 'sentry/components/forms/model';
+import {Field} from 'sentry/components/forms/type';
+import {t} from 'sentry/locale';
+
+type Props = RouteComponentProps<{orgId: string; functionSlug?: string}, {}>;
+// type State = AsyncView['state'] & {
+//   sentryFunction: SentryFunction | null;
+// };
+const formFields: Field[] = [
+  {
+    name: 'name',
+    type: 'string',
+    required: true,
+    placeholder: 'e.g. My Sentry Function',
+    label: 'Name',
+    help: 'Human readable name of your Sentry Function',
+  },
+  {
+    name: 'author',
+    type: 'string',
+    placeholder: 'e.g. Acme Software',
+    label: 'Author',
+    help: 'The company or person who built and maintains this Sentry Function.',
+  },
+];
+
+class SentryFunctionFormModel extends FormModel {
+  getTransformedData(): {} {
+    return super.getTransformedData() as Record<string, any>;
+  }
+}
+
+export default function sentryFunctionDetails(props: Props) {
+  const form = new SentryFunctionFormModel();
```

Comment:
You'll need a `useRef` for this model. Right now each render of this component will make a new form model and erase the current values. If you used `useRef`, instead you'd get the same form model for all render passes.